Sources/NIO/Codec.swift

```diff
@@ -35,6 +35,13 @@ public enum ByteToMessageDecoderError: Error {
     case leftoverDataWhenDone(ByteBuffer)
 }
 
+// TODO (tomer): Merge into ByteToMessageDecoderError next major version
+/// Data errors thrown by `ByteToMessageDecoder`s.
+public enum ByteToMessageDecoderDataError: Error {
+    /// This error can be thrown by `ByteToMessageDecoder`s if the incoming payload is larger than the max specified.
+    case payloadTooLarge
```

Comment:
there's no real point in making this an enum because it has just one case and we can never add a case. This should be a struct:

```swift
public struct ByteToMessageDecoderPayloadTooLargeError: Error {}
```